AdminApi/test/AdminApi.Tests.Integration/Validators/JsonValidators.cs

```diff
@@ -1,28 +1,38 @@
 ﻿using Newtonsoft.Json;
 using Newtonsoft.Json.Linq;
-using Newtonsoft.Json.Schema;
+using NJsonSchema;
 using NJsonSchema.NewtonsoftJson.Generation;
 using JsonSchemaGenerator = NJsonSchema.Generation.JsonSchemaGenerator;
 
 namespace Backbone.AdminApi.Tests.Integration.Validators;
 
 public class JsonValidators
 {
-    private static readonly Dictionary<Type, JSchema> CACHED_SCHEMAS = new();
+    private static readonly Dictionary<Type, JsonSchema> CACHED_SCHEMAS = new();
 
-    public static bool ValidateJsonSchema<T>(string json, out IList<string> errors)
+    public static async Task<(bool IsValid, IList<string> Errors)> ValidateJsonSchema<T>(string json)
     {
+        var errors = new List<string>();
+
         if (CACHED_SCHEMAS.TryGetValue(typeof(T), out var schema))
         {
             try
             {
                 var parsedJson = JObject.Parse(json);
-                return parsedJson.IsValid(schema, out errors);
+                var validationResults = schema.Validate(parsedJson);
+
+                errors.AddRange(validationResults.Select(validationResult => validationResult.ToString()));
+
+                return (validationResults.Count == 0, errors);
```

Comment:
You have repeated this code 4 times in this method. That's not good.

Reply:
I was a bit hesitant to mess too much with the original code, but this was really not good. Refactored.